pkg/container-runtime/installer.go

```diff
@@ -44,39 +49,52 @@ type Installer interface {
 	//Rollback() (ContainerRuntimeInfo, error)
 }
 
-type Config struct {
-	Type         string
-	LimitNofile  string `json:"limitNofile,omitempty" yaml:"limitNofile,omitempty"`
-	CgroupDriver string `json:"cgroupDriver,omitempty" yaml:"cgroupDriver,omitempty"`
-}
-
 type Info struct {
-	Config
+	v2.ContainerRuntimeConfig
+	CgroupDriver   string
 	CRISocket      string
 	CertsDir       string
 	ConfigFilePath string
 }
 
-func NewInstaller(conf Config, driver infradriver.InfraDriver) (Installer, error) {
-	if conf.Type == "docker" {
-		return &DockerInstaller{
+func NewInstaller(conf v2.ContainerRuntimeConfig, driver infradriver.InfraDriver) (Installer, error) {
+	switch conf.Type {
+	case "docker", "":
```

Comment:
could we make "docker","containerd" as a const ? or make it as a valid map["container runtime type"]struct{}, so if another type not found, we cpuld give more detailed error message.